examples/onmt.dynamic.data_config.yaml

```diff
@@ -0,0 +1,41 @@
+# Meta opts:
+## preprocess:
+save_data: generated/sharded.ex0
+overwrite: False
+
+### vocab:
+src_vocab: examples/subword_share/efcjk_shared_50k.spm-onmt_dynamic.vocab
+src_vocab_size: 60000
+tgt_vocab_size: 60000
+vocab_size_multiple: 8
+src_words_min_frequency: 10
+tgt_words_min_frequency: 10
+share_vocab: True
+### Transform related opts:
+src_subword_type: sentencepiece
+tgt_subword_type: sentencepiece
+src_subword_model: examples/subword_share/efcjk_shared_50k.en-fr-zh-ja-ko.spm.model
+tgt_subword_model: examples/subword_share/efcjk_shared_50k.en-fr-zh-ja-ko.spm.model
+subword_nbest: 4
+subword_alpha: 0.1
+onmttok_kwargs: "{'mode': 'aggressive', 'spacer_annotate': True}"
```

Comment:
Can't this be done in a cleaner / indented way?

Reply:
You mean which one? The `onmttok_kwargs` ?

Reply:
Yes.

Reply:
The thing is there are lots of arguments in OpenNMT Tokenizer. It will make the config a mess to adding all of them into the `opt.py` while raises dependency on theirs. The `onmttok_kwargs` tend to pass all arguments unrelated to subword to Tokenizer therefore no dependency while robust to changes.

Reply:
Hmmm makes sense.
I'm really not fond of this "dict string" though.
What would you think about adding a new config file for the tokenizer(s), which we would parse and convert to a kwargs dict for pyonmttok? (Maybe I'm going a bit too far here...)

Reply:
Also, in the same topic, BPE dropout has been added to Tokenizer (https://github.com/OpenNMT/Tokenizer/pull/157) with the new `bpe_dropout` kwarg, so we should see if we want to add it explicitly or handle it via the dict string.

Reply:
> Hmmm makes sense.
> I'm really not fond of this "dict string" though.
> What would you think about adding a new config file for the tokenizer(s), which we would parse and convert to a kwargs dict for pyonmttok? (Maybe I'm going a bit too far here...)

Adding another config file makes it more complicated and confusing, isn't it? If we really want to do something like this, maybe add that to nmt_pipeline would be better. And as for me, the Tokenizer Transform is an option not compulsory, I can image the users may use mose_tokenize + sentencepiece/bpe regularization instead. Therefore, the "dict string" is simplest to not disturb those who don't use them, and for those who use this option, they should know what they are doing anyway.

Reply:
> Also, in the same topic, BPE dropout was added to Tokenizer ([OpenNMT/Tokenizer#157](https://github.com/OpenNMT/Tokenizer/pull/157)) with the new `bpe_dropout` kwarg, so we should see if we want to add it explicitly or handle it via the dict string.

Yes. The subword part is intended to be handled explicitly. And I may prefer to unify the `bpe_dropout` to `subword_alpha` if it's not given in the kwargs, otherwise pass it to Tokenizer directly. This makes sure our sentencepiece port and Tokenizer port are unified in both cases.

Reply:
Makes sense. Enabled this in 3248756 in #4 . (Does not handle the "if it's not given in the kwargs, otherwise pass it to Tokenizer directly" though, as if it's repeated it will raise a `SyntaxError` anyways.)
By the way, it seems the dropout was inverted (1 - alpha), but from what I read it should not (alpha = dropout prob). Did I miss something?

Reply:
> Makes sense. Enabled this in [3248756](https://github.com/Zenglinxiao/OpenNMT-py/commit/32487562dbd25dca38502dc32819a74da226e991) in #4 . (Does not handle the "if it's not given in the kwargs, otherwise pass it to Tokenizer directly" though, as if it's repeated it will raise a `SyntaxError` anyways.)
> By the way, it seems the dropout was inverted (1 - alpha), but from what I read it should not (alpha = dropout prob). Did I miss something?

I double-checked the mentioned `alpha` in sentencepiece:
* In [sentencepiece_processor.h](https://github.com/google/sentencepiece/blob/a2c12f038fd4a572c0d8ab14ff1850625bae97d7/src/sentencepiece_processor.h#L288): "// `alpha` is the merge probability `p` in BPE-dropout paper.", which make me think it should be the reverse of dropout.
* But in the [bpe_model.h](https://github.com/google/sentencepiece/blob/856daadbbfbf26da81152e70aba0406a11d5bedc/src/bpe_model.h#L41):  // "Skips merge operation with `alpha` probability.".

So, yes, `alpha` = `dropout`, the description of sentencepiece is confusing.